flutter_local_notifications_windows/src/plugin.cpp

```diff
@@ -0,0 +1,225 @@
+#include <sstream>
+
+#include <windows.h>  // <-- This must be the first Windows header
+#include <appmodel.h>
+#include <atlbase.h>
+#include <NotificationActivationCallback.h>
+#include <VersionHelpers.h>
+
+#include "plugin.hpp"
+#include "utils.hpp"
+
+struct RegistryHandle {
+	using type = HKEY;
+
+	static void close(type value) noexcept {
+		WINRT_VERIFY_(ERROR_SUCCESS, RegCloseKey(value));
+	}
+
+	static constexpr type invalid() noexcept { return nullptr; }
+};
+
+using RegistryKey = winrt::handle_type<RegistryHandle>;
+
+/// This callback will be called when a notification sent by this plugin is clicked on.
+struct NotificationActivationCallback : winrt::implements<NotificationActivationCallback, INotificationActivationCallback> {
+	NativeNotificationCallback callback;
+
+	HRESULT __stdcall Activate(LPCWSTR app, LPCWSTR args, NOTIFICATION_USER_INPUT_DATA const* data, ULONG count) noexcept final {
+		try {
```

Comment:
A very minor thing I noticed just now. Looks like the formatting here is off? I don't work with C++ but from what I read [ClangFormat](https://clang.llvm.org/docs/ClangFormat.html) can be applied. If so, would it be possible to include in this PR an update to [this workflow](https://github.com/MaikuB/flutter_local_notifications/blob/master/.github/workflows/format.yml) so that it could format C++ plugin introduced by the Windows implementation too?

Reply:
I set up clang-format with a custom `.clang-format` file, designed to make minimal changes while enforcing consistency.

Reply:
Note: I used a pip-based installation of clang (19) for now since `ubuntu-latest` is `22.04` which can't download `clang-format-17` (jammy vs noble). I only now saw that you can manually select `ubuntu-24.0` in the workflow, which might work, but I don't have time to test that right now.

Reply:
Thanks have a comment but I'll leave it to where it's more appropriate and leave this one as resolved